improver/ensemble_copula_coupling/utilities.py

```diff
@@ -308,19 +308,28 @@ def slow_interp_same_y(x: np.ndarray, xp: np.ndarray, fp: np.ndarray) -> np.ndar
     return result
 
 
-try:
-    import numba  # noqa: F401
-
-    from improver.ensemble_copula_coupling.numba_utilities import fast_interp_same_y
+def interpolate_multiple_rows_same_y(*args):
+    """For each row i of xp, do the equivalent of np.interp(x, xp[i], fp).
 
-    def interpolate_multiple_rows_same_y(*args):
-        return fast_interp_same_y(*args)
+    Calls a fast numba implementation where numba is available (see
+    `improver.ensemble_copula_coupling.numba_utilities.fast_interp_same_y`) and calls a
+    the native python implementation otherwise (see :func:`slow_interp_same_y`).
 
+    Args:
+        x: 1-d array
+        xp: n * m array, each row must be in non-decreasing order
+        fp: 1-d array with length m
+    Returns:
+        n * len(x) array where each row i is equal to np.interp(x, xp[i], fp)
+    """
+    try:
+        import numba  # noqa: F401
```

Comment:
Will importing numba each time the function is called add a lot of overhead?

Reply:
Shouldn't do no.  After the first time, python does a lookup of the already imported numba internally.

Reply:
Ah ok, I didn't realise that. Seems fine then!